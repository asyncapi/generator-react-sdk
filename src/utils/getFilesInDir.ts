
import Path from 'path';
import fs from 'fs';
import { promisify } from 'util';
const readdir = promisify(fs.readdir);
const stat = promisify(fs.stat);

class GetFilesResponse{
    files: string[];
    dirs: string[];
    constructor(files: string[], dirs: string[]){
        this.files = files;
        this.dirs = dirs;
    }
}

/**
 * Function which finds all the files in directories
 * 
 * @param dir directory to find files in.
 * @param includeSubDirs should the function iterate through subdirectories to search for files?
 */
export async function getFilesInDir(dir: string){
    const allFiles = await readdir(dir);
    let files: string[] = [];
    let dirs: string[] = [];
    for (const filename of allFiles) {
        const res = Path.resolve(dir, filename);
        const isDir = (await stat(res)).isDirectory();
        if(isDir){
            dirs.push(res)
        }else{
            files.push(res)
        }
    }
    const resolveFilenameCallback = (filename: string) => {
        return Path.resolve(dir, filename); 
    }
    files = files.map(resolveFilenameCallback);
    dirs = dirs.map(resolveFilenameCallback);
    return new GetFilesResponse(files, dirs);
}