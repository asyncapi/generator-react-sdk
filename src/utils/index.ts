export * from "./getFilesInDir";
export * from "./isJsFile";
export * from "./withIndendation";
export * from "./withNewLines";
