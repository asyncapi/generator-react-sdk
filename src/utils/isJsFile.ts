const ALLOWED_EXTS = ['js', 'jsx', 'cjs', 'tsx'];

/**
 * Function which checks if file is JS file that supports JSX syntax
 * 
 * @private
 * @param {string} filename
 * @returns {boolean}
 */
export function isJsFile(filename: string = '') {
  const ext = filename.split('.').pop() || '';
  return ALLOWED_EXTS.includes(ext);
}
