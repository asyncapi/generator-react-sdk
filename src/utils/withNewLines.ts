/**
 * 
 * @private
 * @param {string} content 
 * @param {number} number 
 */
export function withNewLines(content: string = '', number: number = 0) {
  const newLines = Array(number).fill('\n').join('');
  return content + newLines;
}
