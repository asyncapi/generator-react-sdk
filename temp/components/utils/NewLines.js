/**
 * 
 * @private
 * @param {*} content 
 * @param {*} number 
 */
export function withNewLines(content = '', number = 0) {
  const newLines = Array(number).fill('\n').join('');
  return content + newLines;
}