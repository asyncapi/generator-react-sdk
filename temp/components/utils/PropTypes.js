import PropTypes from 'prop-types';
import { IndentationTypes } from '../enums/IndentationTypes';
export const IndentationPropTypes = {
  /**
   * `size` prop expects a string which format should be a number which represent the number of `type`'s to indent each content. Default to no indentation.
   */
  size: PropTypes.string,
  /**
   * `type` prop expects a string, you can use the `IndentationTypes` enum, as either `TABS` or `SPACES`. The `type` defaults to `SPACES`. 
   */
  type: PropTypes.oneOf(Object.values(IndentationTypes)),
};