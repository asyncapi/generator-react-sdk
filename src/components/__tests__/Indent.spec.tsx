import React from 'react';
import { IndentationTypes } from '../../utils';
import { Indent, render, Text} from "../..";

describe('<Indent />', () => {
  test('Should always render indentation as is', () => {
    const defaultProps = {};
    const wrapper = render(<Indent {...defaultProps}>Test</Indent>);
    expect(wrapper).toMatchSnapshot('indent_should_render');
  });
  test('Should always render indentation with spaces', () => {
    const defaultProps = {
      size: 4,
      type: IndentationTypes.SPACES
    };
    const wrapper = render(<Indent {...defaultProps}>Test</Indent>);
    expect(wrapper).toMatchSnapshot('indent_should_render_space_indentation');
  });
  test('Should always render indentation with tabs', () => {
    const defaultProps = {
      size: 4,
      type: IndentationTypes.TABS
    };
    const wrapper = render(<Indent {...defaultProps}>Test</Indent>);
    expect(wrapper).toMatchSnapshot('indent_should_render_tab_indentation');
  });
  test('Should be able to nest indentation', () => {
    const defaultProps = {
      size: 4,
      type: IndentationTypes.TABS
    };
    const wrapper = render(<Indent {...defaultProps}><Indent {...defaultProps}>Test</Indent></Indent>);
    expect(wrapper).toMatchSnapshot('indent_should_render_nested_indentation');
  });
  test('Should be able contain text component', () => {
    const defaultProps = {
      size: 4,
      type: IndentationTypes.TABS
    };
    const defaultTextProps = {
      indent: 4,
      type: IndentationTypes.TABS
    };
    const wrapper = render(<Indent {...defaultProps}><Text {...defaultTextProps}>Test</Text></Indent>);
    expect(wrapper).toMatchSnapshot('indent_should_render_text_component');
  });
});