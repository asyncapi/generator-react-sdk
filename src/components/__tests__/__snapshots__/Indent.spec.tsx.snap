// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Indent /> Should always render indentation as is: indent_should_render 1`] = `"Test"`;

exports[`<Indent /> Should always render indentation with spaces: indent_should_render_space_indentation 1`] = `"    Test"`;

exports[`<Indent /> Should always render indentation with tabs: indent_should_render_tab_indentation 1`] = `"				Test"`;

exports[`<Indent /> Should be able contain text component: indent_should_render_text_component 1`] = `
"								Test
"
`;

exports[`<Indent /> Should be able to nest indentation: indent_should_render_nested_indentation 1`] = `"								Test"`;
